thirdparty/dxil_mach/MachSiegbertVogtDXCSA.cpp

```diff
@@ -0,0 +1,234 @@
+// Licensed under the Mach engine license (Apache or MIT at your choosing.) For details
+// and a copy of this license, see https://github.com/hexops/mach/blob/main/LICENSE
+// This copyright header, and a copy of the above open source licenses, must be provided
+// with any redistributions of this software.
+
+#include <string.h>
+#include "MachSiegbertVogtDXCSA.h"
+
+// Note: If this algorithm looks familiar to you, it is - but not exactly. Fun!
```

Comment:
Someone should analyze this and determine what's the difference between this magic and MD5, but I'm too lazy to do it.